### PR TITLE
fix: Update docs for Python SDK 0.9.0

### PR DESCRIPTION
**Pull request by untitaker**
The logic to determine what errors should be reported is unusually complex for Pyramid. Let me know if you have a better way to structure this entire page @MimiDumpling, I tend to make too many bulletpoint lists when my sentences get too long.

### CODE REVIEW

File: src/collections/_documentation/platforms/python/pyramid.md [hunk 1]
```diff
@@ -39,7 +39,10 @@ Framework](https://trypyramid.com/).
 
 * The Sentry Python SDK will install the Pyramid integration for all of your apps. The integration hooks into Pyramid itself, not any of your apps specifically.
 
-* All exceptions leading to an Internal Server Error are reported.
+* All exceptions leading to an Internal Server Error are reported. This means exceptions that:
```
Comment (MimiDumpling) [inline src/collections/_documentation/platforms/python/pyramid.md]:
```suggestion
*The SDK will report all exceptions leading to an Internal Server Error. These two kinds of exceptions are:
```

File: src/collections/_documentation/platforms/python/pyramid.md [hunk 2]
```diff
@@ -39,7 +39,10 @@ Framework](https://trypyramid.com/).
...
+
+  * are not handled by any exception view, or
```
Comment (MimiDumpling) [inline src/collections/_documentation/platforms/python/pyramid.md]:
```suggestion
  * exceptions that are not handled by any exception view
```

File: src/collections/_documentation/platforms/python/pyramid.md [hunk 3]
```diff
@@ -39,7 +39,10 @@ Framework](https://trypyramid.com/).
...
+  * whose exception view returns a status code of 500 (Pyramid version 1.9+ only)
```
Comment (MimiDumpling) [inline src/collections/_documentation/platforms/python/pyramid.md]:
```suggestion
  * exceptions whose exception view returns a status code of 500 (Pyramid version 1.9+ only)
```

**Comment by untitaker** [comment]
Thanks for the review @MimiDumpling, please review again (merge blocked on approval)

**Pull request merged**